src/Underworld.ts

```diff
@@ -2373,23 +2373,25 @@ ${CardUI.cardListToImages(player.stats.longestSpell)}
   // Sends a network message to end turn
   async endMyTurnButtonHandler() {
     if (globalThis.player) {
-      // If End turn button is level up button
-      if (elEndTurnBtn.classList.contains('upgrade') || document.body?.classList.contains(showUpgradesClassName)) {
-        const upgradesLeftToChoose = this.upgradesLeftToChoose(globalThis.player)
-        if (upgradesLeftToChoose > 0) {
-          this.showUpgrades();
-          elEndTurnBtn.classList.toggle('upgrade', false);
-          // Do not end turn, just show upgrades
+      if (elEndTurnBtn) {
+        // If End turn button is level up button
+        if (elEndTurnBtn.classList.contains('upgrade') || document.body?.classList.contains(showUpgradesClassName)) {
```

Comment:
small nitpick: This could be rewritten as `if(elEndTurnBtn && elEndTurnBtn.classList...` to reduce the nested if statements

Reply:
No problem, happy to make it look cleaner. This is fixed in 44b92cb